main.tf

```diff
@@ -170,24 +86,21 @@ module "container_definition" {
 }
 
 module "redis" {
-  count   = var.redis_enabled ? 1 : 0
+  count   = module.redis_label.enabled ? 1 : 0
   source  = "cloudposse/ecs-alb-service-task/aws"
   version = "0.66.4"
 
-  context = module.redis_label[0].context
+  context = module.redis_label.context
```

Comment:
i don't see why we should have the redis_label module in the state if we'd set var.redis_enabled=false.